nidx/nidx_text/tests/test_search.rs

```diff
@@ -307,4 +353,101 @@ fn test_int_order_pagination() {
     assert!(response.next_page);
 }
 
-// TODO: order, timestamp filter, only_faceted, with_status
+#[test]
+fn test_timestamp_filtering() {
+    let reader = common::test_reader();
+
+    let now = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH).unwrap();
+    let before = Timestamp {
+        seconds: now.as_secs() as i64 - 100,
+        nanos: 0,
+    };
+    let after = Timestamp {
+        seconds: now.as_secs() as i64 + 100,
+        nanos: 0,
+    };
+
+    let search = |date_range| {
+        let request = DocumentSearchRequest {
+            id: "shard".to_string(),
+            body: "".to_string(),
+            page_number: 0,
+            result_per_page: 10,
+            min_score: f32::MIN,
+            filter_expression: Some(FilterExpression {
+                expr: Some(Expr::Date(date_range)),
+            }),
+            ..Default::default()
+        };
+        let response = reader.search(&request).unwrap();
+        response.results.len()
+    };
+
+    for field in &[DateField::Created.into(), DateField::Modified.into()] {
+        assert_eq!(
+            search(DateRangeFilter {
+                field: *field,
+                since: Some(before),
+                until: Some(after)
+            }),
+            2
+        );
+
+        assert_eq!(
+            search(DateRangeFilter {
+                field: *field,
+                since: Some(after),
+                until: None
+            }),
+            0
+        );
+    }
+}
+
+#[test]
+fn test_key_filtering() {
+    let reader = common::test_reader();
+
+    let request = DocumentSearchRequest {
+        id: "shard".to_string(),
+        body: "".to_string(),
+        page_number: 0,
+        result_per_page: 1,
+        min_score: f32::MIN,
+        ..Default::default()
+    };
+    let response = reader.search(&request).unwrap();
+    let resource_id = &response.results[0].uuid;
+
+    let search = |resource| {
+        let request = DocumentSearchRequest {
+            id: "shard".to_string(),
+            body: "".to_string(),
+            page_number: 0,
+            result_per_page: 10,
+            min_score: f32::MIN,
+            filter_expression: Some(FilterExpression {
+                expr: Some(Expr::Resource(resource)),
+            }),
+            ..Default::default()
+        };
+        let response = reader.search(&request).unwrap();
+        response.results.len()
+    };
+
+    // By resource
+    assert_eq!(
+        search(ResourceFilter {
+            resource_id: resource_id.clone(),
+        }),
+        2
+    );
+    assert_eq!(
+        search(ResourceFilter {
+            resource_id: "fake".into(),
+        }),
+        0
+    );
+}
+
+// TODO: order, only_faceted, with_status
```

Comment:
with_status only apply to catalog right?

Reply:
Yeah, in the nucliadb PR I'm removing it.